My Site Dashboard: Hide segmented control for self-hosted sites

momo-ozawa opened the pull request:
Fixes #17825 

## Description

This PR hides the segmented control on the My Site screen for:
- self-hosted sites w/o Jetpack
- iPad devices

This PR also adds a call to the `showInitialDetailsForBlog` method when switching sites via the site picker. This call ensures that the correct details vc will be shown when switching sites on an iPad.

## How to test

Make sure the My Site Dashboard feature flag is **enabled**.

### Self-hosted sites: Flow 1
1. Log out
2. Login into a **self-hosted site w/o Jetpack**
3. ✅ The segmented control should be hidden
4. Close the app then reopen it

<details>
  <summary>Click to expand for video</summary>
  
  https://user-images.githubusercontent.com/6711616/151529512-b6fe85e9-e91f-438a-b271-dcfd07945a60.mp4
</details>

### Self-hosted sites: Flow 2

1. Log out
2. Log into with your WP.com account
3. Switch to a **WP.com** site
4. ✅ The segmented control should be visible
5. Switch to an **Atomic** site
6. ✅ The segmented control should be visible
7. Switch to a **self-hosted site with Jetpack**
8. ✅ The segmented control should be visible
9. On the segmented control, switch to the **dashboard**
10. Switch to a **self-hosted site w/o Jetpack**
11. ✅ The segmented control should be hidden and the site menu should be displayed
12. Switch to a WP.com/Atomic/Jetpack site
13. ✅ The segmented control should be visible

<details>
  <summary>Click to expand for video</summary>
  
  https://user-images.githubusercontent.com/6711616/151529294-97c59a86-16f8-4ba5-a780-61ea1c801344.mp4
</details>

### iPads
1. Log out
2. Log in with your WP.com account
3. ✅ The segmented control should be hidden
4. Close the app then reopen it
5. ✅ The segmented control should be hidden
6. Select a different site on the site switcher
7. ✅ The detail vc should reflect the newly selected site

<details>
  <summary>Click to expand for video</summary>

  https://user-images.githubusercontent.com/6711616/151532084-fc2fc2e6-d3b2-4ea8-a5e4-dff8a82fd2f9.mp4

</details>


## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [X] I have completed the Regression Notes.
- [X] I have considered adding unit tests for my changes.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Refactor: make sure segemnted control is hidden for self-hosted sites

Commit:
Refactor: hide segmented control if device is an iPad

leandroalonso commented on WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift:
```diff
@@ -149,16 +151,12 @@ class MySiteViewController: UIViewController, NoResultsViewHost {
+            return
         }
-        segmentedControl.selectedSegmentIndex = 0
+        segmentedControlContainerView.isHidden = !blog.isHostedAtWPcom || UIDevice.isPad()
```
`blog.isHostedAtWPcom` doesn't take into account Jetpack connected sites (including Atomic), and the dashboard is supported for them.

momo-ozawa replied:
Fixed! 46a0d73

I added these use cases to the testing instructions above, as well as a video.

leandroalonso reviewed (commented):
@momo-ozawa we have a few more users cases to take into account. Although unlikely, a user can be logged in to WP.com and have a self-hosted site added. These are the users' cases I can think of that are not supported:

### Jetpack connected site

1. When I select (or open the app) with a Jetpack connected site
2. Then I see the segmented control

### Atomic

1. When I select (or open the app) with an Atomic
2. Then I see the segmented control

### Changing blogs

1. When I have a WP.com/Atomic/Jetpack site selected
2. I see the segmented control
3. But when I tap the site picker and select a self-hosted user
4. The segmented control is hidden
5. And if the Dashboard screen is selected, the app switch to the blog details

Ps.: for this last one, we have also the opposite, from a self-hosted to a WP.com/Atomic/Jetpack site (thus showing the segmented control)

Commit:
Fix: segmented control logic

The segmented control should be hidden if the blog is NOT a WP.com/Atomic/Jetpack site, or if the device is an iPad.

Also, when switching blogs from a WP.com/Atomic/Jetpack site to a self-hosted site (without Jetpack) and if the dashboard is showing, we need to toggle the segmented control to display the site menu.

leandroalonso commented on WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift:
```diff
@@ -557,6 +557,14 @@ class MySiteViewController: UIViewController, NoResultsViewHost {
 
         sitePickerViewController.onBlogSwitched = { [weak self] blog in
+
+            let isShowingDashboard = self?.segmentedControl.selectedSegmentIndex == Section.dashboard.rawValue
```
Nit: we can extract `isShowingDashboard` to its own var (it's probably going to be useful in the future).

momo-ozawa replied:
Fixed! 40ce517

leandroalonso reviewed (approved):
:shipit:

Commit:
Refactor: extract isShowingDashboard to a var
